### PR TITLE
Ddcnl 8952

### PR DESCRIPTION
**Pull request by grantrobson**
- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -41,7 +41,7 @@ lazy val microservice = Project(appName, file("."))
       "-unchecked",
       "-feature",
       "-Xlint:_",
-      "-Werror",
+//      "-Werror",
```
Comment (pascal22p) [inline build.sbt:44]: This needs to be put back

File: app/uk/gov/hmrc/tai/util/SensitiveHelper.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  private def writesSensitiveJsValue(implicit crypto: Encrypter): Writes[SensitiveJsValue] = { sjo: SensitiveJsValue =>
+    JsString(crypto.encrypt(PlainText(Json.stringify(sjo.decryptedValue))).value)
+  }
+
+  private def readsSensitiveJsValue[A <: JsValue: Format](implicit crypto: Decrypter): Reads[SensitiveJsValue] = {
+    case JsString(s) =>
+      Try(crypto.decrypt(Crypted(s))) match {
+        case Success(plainText)            => JsSuccess(SensitiveJsValue(Json.parse(plainText.value).as[A]))
+        case Failure(_: SecurityException) => JsSuccess(SensitiveJsValue(JsString(s).as[A]))
```
Comment (pascal22p) [inline app/uk/gov/hmrc/tai/util/SensitiveHelper.scala:35]: I think this is the fallback if decryption fails? Maybe a comment on why we do this for future generation

File: app/uk/gov/hmrc/tai/connectors/EmploymentDetailsConnector.scala [hunk 3]
```diff
@@ -31,17 +33,20 @@ import scala.concurrent.Future
...
+  crypto: ApplicationCrypto
 ) extends EmploymentDetailsConnector {
 
   override val originatorId: String = config.originatorId
   override val baseUrl: String = config.baseURL
   override def getEmploymentDetailsAsEitherT(nino: Nino, year: Int)(implicit
     hc: HeaderCarrier
-  ): EitherT[Future, UpstreamErrorResponse, HodResponse] =
+  ): EitherT[Future, UpstreamErrorResponse, HodResponse] = {
+    implicit val encrypterDecrypter: Encrypter with Decrypter = crypto.JsonCrypto
```
Comment (pascal22p) [inline app/uk/gov/hmrc/tai/connectors/EmploymentDetailsConnector.scala:45]: this crypto points to crypto library instead of our own CryptoProvider?

**Review by pascal22p — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/tai/connectors/EmploymentDetailsConnector.scala (hunk 3 above)
Comment (grantrobson) [reply app/uk/gov/hmrc/tai/connectors/EmploymentDetailsConnector.scala:45]: I think it's the same one. I just added a CryptoProvider but it points to the same crypto library.

File: app/uk/gov/hmrc/tai/util/SensitiveHelper.scala (hunk 2 above)
Comment (grantrobson) [reply app/uk/gov/hmrc/tai/util/SensitiveHelper.scala:42]: Added comment

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

**Review by pascal22p — APPROVED** [review]

**Pull request merged**